runtime/wasm/src/host.rs

```diff
@@ -107,7 +106,9 @@ impl RuntimeHost {
             // Load the mappings as a WASM module
             let module = WasmiModule::new(&logger, wasmi_config);
 
-            Self::subscribe_to_events(&logger, data_source, module, ethereum_adapter);
+            Self::subscribe_to_events(&logger, data_source, module, ethereum_adapter)
+                .wait()
```

Comment:
I find moving this wait + for_each out of subscribe_to_events odd and would prefer this to be part of the function.

Reply:
The bug was caused because of `wait` being called on multiple streams, now it's very visible that we wait on a single stream. So I prefer this form, even though this line doesn't read very well, I've added a comment to attempt to clarify what's happening, what do you think?